### ISSUE TITLE
Regarding anonymous login issue

### ISSUE THREAD

**Issue by dongwoo-john-ku** [elastic/Machinebeat#21]
Hello,

I'm writing to inquire about 'anonymous login' issue.
First of all, thank you for making machinebeat.

It looks a nice tool and I succeed to connect to the demo endpoint using your program without any problem.
 * demo endpoint: opc.tcp://opcuaserver.com:48010

However, I have a problem when I connect to my OPC UA server.
My local endpoint information like below and I checked the OPC UA server has no problem with another OPC/UA client tools 
 * local endpoint: opc.tcp://host.docker.internal:53530/OPCUA/SimulationServer

Is there some configuration what I can do?
Here is disconnection full log information and thank you in advanced. 

```
2020-08-11T17:44:35.246+0900    INFO    instance/beat.go:622    Home path: [C:\Users\dwku\OneDrive\바탕 화면\machinebeat] Config path: [C:\Users\dwku\OneDrive\바탕 화면\machinebeat] Data path: [C:\Users\dwku\OneDrive\바탕 화면\machinebeat\data] Logs path: [C:\Users\dwku\OneDrive\바탕 화면\machinebeat\logs]
2020-08-11T17:44:35.307+0900    INFO    instance/beat.go:630    Beat ID: d51d9d17-d145-489f-b43d-f9953bc4d771
2020-08-11T17:44:35.355+0900    INFO    [beat]  instance/beat.go:958    Beat info       {"system_info": {"beat": {"path": {"config": "C:\\Users\\dwku\\OneDrive\\바탕 화면\\machinebeat", "data": "C:\\Users\\dwku\\OneDrive\\바탕 화면\\machinebeat\\data", "home": "C:\\Users\\dwku\\OneDrive\\바탕 화면\\machinebeat", "logs": "C:\\Users\\dwku\\OneDrive\\바탕 화면\\machinebeat\\logs"}, "type": "machinebeat", "uuid": "d51d9d17-d145-489f-b43d-f9953bc4d771"}}}
2020-08-11T17:44:35.356+0900    INFO    [beat]  instance/beat.go:967    Build info      {"system_info": {"build": {"commit": "unknown", "libbeat": "7.6.1", "time": "1754-08-30T22:43:41.128Z", "version": "7.6.1"}}}
2020-08-11T17:44:35.356+0900    INFO    [beat]  instance/beat.go:970    Go runtime info {"system_info": {"go": {"os":"windows","arch":"amd64","max_procs":8,"version":"go1.12.4"}}}
2020-08-11T17:44:35.358+0900    INFO    add_cloud_metadata/add_cloud_metadata.go:89     add_cloud_metadata: hosting provider type not detected.
2020-08-11T17:44:35.402+0900    INFO    [beat]  instance/beat.go:974    Host info       {"system_info": {"host": {"architecture":"x86_64","boot_time":"2020-08-11T13:44:01.31+09:00","name":"SEOUL-DKU","ip":["fe80::cc0f:37ad:34f1:3009/64","169.254.48.9/16","fe80::90e8:fdde:a2e0:8875/64","169.254.136.117/16","fe80::49f8:3301:4b72:98ad/64","192.168.0.6/24","fe80::a4fd:5499:5ae7:de0a/64","169.254.222.10/16","::1/128","127.0.0.1/8","fe80::4928:1421:1cf2:2d4b/64","172.25.129.17/28"],"kernel_version":"10.0.17134.1610 (WinBuild.160101.0800)","mac":["92:32:4b:1c:90:47","d2:32:4b:1c:90:47","90:32:4b:1c:90:47","90:32:4b:1c:90:48","72:15:91:d4:83:ca"],"os":{"family":"windows","platform":"windows","name":"Windows 10 Enterprise","version":"10.0","major":10,"minor":0,"patch":0,"build":"17134.1610"},"timezone":"KST","timezone_offset_sec":32400,"id":"e47d409c-3cca-409f-a573-66904f0b01e3"}}}
2020-08-11T17:44:35.405+0900    INFO    [beat]  instance/beat.go:1003   Process info    {"system_info": {"process": {"cwd": "C:\\Users\\dwku\\OneDrive\\바탕  화면\\machinebeat", "exe": "C:\\Users\\dwku\\OneDrive\\\ufffd\ufffd\ufffd\ufffd ȭ\ufffd\ufffd\\machinebeat\\machinebeat.exe", "name": "machinebeat.exe", "pid": 22124, "ppid": 22836, "start_time": "2020-08-11T17:44:34.360+0900"}}}
2020-08-11T17:44:35.406+0900    INFO    instance/beat.go:298    Setup Beat: machinebeat; Version: 7.6.1
2020-08-11T17:44:35.406+0900    INFO    [index-management]      idxmgmt/std.go:182      Set output.elasticsearch.index to 'machinebeat-7.6.1' as ILM is enabled.
2020-08-11T17:44:35.407+0900    INFO    elasticsearch/client.go:174     Elasticsearch url: http://localhost:9200
2020-08-11T17:44:35.407+0900    INFO    [publisher]     pipeline/module.go:110  Beat name: SEOUL-DKU
2020-08-11T17:44:35.408+0900    INFO    instance/beat.go:439    machinebeat start running.
2020-08-11T17:44:35.408+0900    INFO    [monitoring]    log/log.go:118  Starting metrics logging every 30s
2020-08-11T17:44:35.410+0900    WARN    [cfgwarn]       nodevalue/nodevalue.go:92       BETA: The OPCUA metricset is beta.
2020-08-11T17:44:35.410+0900    INFO    nodevalue/client.go:54  [OPCUA] Get all endpoints from opc.tcp://host.docker.internal:53530/OPCUA/SimulationServer
2020-08-11T17:44:35.432+0900    INFO    nodevalue/client.go:68  [OPCUA] Policy URI: http://opcfoundation.org/UA/SecurityPolicy#None with security mode MessageSecurityModeNone
2020-08-11T17:44:35.432+0900    INFO    nodevalue/client.go:88  [OPCUA] Set to anonymous login
2020-08-11T17:44:36.449+0900    ERROR   nodevalue/nodevalue.go:165      [OPCUA] Tried to connect to OPC UA server 1 time(s). Without success.
2020-08-11T17:44:36.462+0900    INFO    [monitoring]    log/log.go:153  Total non-zero metrics  {"monitoring": {"metrics": {"beat":{"cpu":{"system":{"ticks":156,"time":{"ms":156}},"total":{"ticks":187,"time":{"ms":187},"value":187},"user":{"ticks":31,"time":{"ms":31}}},"handles":{"open":236},"info":{"ephemeral_id":"bd0cdfe6-5384-4376-9e79-4223acf6bc67","uptime":{"ms":2072}},"memstats":{"gc_next":5973680,"memory_alloc":3132680,"memory_total":6589824,"rss":34734080},"runtime":{"goroutines":17}},"libbeat":{"config":{"module":{"running":0}},"output":{"type":"elasticsearch"},"pipeline":{"clients":0,"events":{"active":0}}},"system":{"cpu":{"cores":8}}}}}
2020-08-11T17:44:36.462+0900    INFO    [monitoring]    log/log.go:154  Uptime: 2.0728753s
2020-08-11T17:44:36.463+0900    INFO    [monitoring]    log/log.go:131  Stopping metrics logging.
2020-08-11T17:44:36.464+0900    INFO    instance/beat.go:445    machinebeat stopped.
2020-08-11T17:44:36.465+0900    ERROR   instance/beat.go:933    Exiting: 1 error: Connection was not possible
Exiting: 1 error: Connection was not possible
```

**Comment by felix-lessoer** [comment]
Are you sure your docker endpoint is supporting anonymous login without any certificate?
Could you also send DEBUG level logs?

**Comment by dongwoo-john-ku** [comment]
Hello @felix-lessoer, 
Thank you for your quick reply.

Yes, the endpoint is supporting anonymous login and I could log in the endpoint via uaExpert and a Python library( without any certificate.

I made the endpoint via Prosys OPC UA Simulation Server
![image](https://user-images.githubusercontent.com/61130023/89892296-6711dc80-dc11-11ea-8224-557ffd3ca5d2.png)

Please refer the below log from exExpert.
`20:26:00.264 | UaExpert           |                                | Loaded Data Access Plugin (Static Plugin)
20:26:00.264 | UaExpert           |                                | Loaded History Plugin (Static Plugin)
20:26:00.264 | UaExpert           |                                | Loaded Events Plugin (Static Plugin)
20:26:00.264 | UaExpert           |                                | Loaded Method Plugin (Static Plugin)
20:26:00.264 | UaExpert           |                                | Loaded Performance Plugin (Static Plugin)
20:26:00.264 | UaExpert           |                                | Loaded Image Viewer Plugin (Static Plugin)
20:26:00.264 | UaExpert           |                                | Loaded Server Diagnostic Plugin (Static Plugin)
20:26:00.264 | UaExpert           |                                | Loaded Data Logger Plugin (Static Plugin)
20:26:00.264 | UaExpert           |                                | Loaded GDS Push Plugin (Static Plugin)
20:26:00.264 | UaExpert           |                                | Loaded File Transfer Plugin (Static Plugin)
20:26:00.264 | UaExpert           |                                | Loaded XML Nodeset Export Plugin (Static Plugin)
20:26:00.275 | UaExpert           |                                | UaExpert is ready to use.
20:26:34.321 | DiscoveryWidget    |                                | Discovery FindServersOnNetwork on opc.tcp://localhost:4840 failed (BadTimeout), falling back to FindServers
20:26:34.824 | DiscoveryWidget    |                                | Discovery FindServers on opc.tcp://localhost:4840 failed (BadTimeout)
20:26:35.326 | DiscoveryWidget    |                                | Discovery GetEndpoints on opc.tcp://localhost:4840 failed (BadTimeout)
20:26:38.857 | Server Node        | opc.tcp://opcuaserver.com:4... | Endpoint: 'opc.tcp://host.docker.internal:53530/OPCUA/SimulationServer'
20:26:38.857 | Server Node        | opc.tcp://opcuaserver.com:4... | Security policy: 'http://opcfoundation.org/UA/SecurityPolicy#None'
20:26:38.857 | Server Node        | opc.tcp://opcuaserver.com:4... | ApplicationUri: 'urn:host.docker.internal:OPCUA:SimulationServer'
20:26:38.857 | Server Node        | opc.tcp://opcuaserver.com:4... | Used UserTokenType: Anonymous
20:26:39.122 | AddressSpaceModel  | opc.tcp://opcuaserver.com:4... | Registered for ModelChangeEvents
20:26:39.122 | Server Node        | opc.tcp://opcuaserver.com:4... | Connection status of server 'opc.tcp://opcuaserver.com:48010' changed to 'Connected'.
20:26:39.122 | Server Node        | opc.tcp://opcuaserver.com:4... | Revised values: SessionTimeout=1200000, SecureChannelLifetime=3600000
20:26:39.126 | AddressSpaceModel  | opc.tcp://opcuaserver.com:4... | Browse on node 'i=84' succeeded.
20:26:39.131 | TypeCache          | opc.tcp://opcuaserver.com:4... | Reading type info of NodeId NS0|Numeric|35 succeeded
20:26:39.137 | TypeCache          | opc.tcp://opcuaserver.com:4... | Reading type info of NodeId NS0|Numeric|33 succeeded
20:26:39.146 | TypeCache          | opc.tcp://opcuaserver.com:4... | Reading type info of NodeId NS0|Numeric|31 succeeded
20:26:39.154 | AddressSpaceModel  | opc.tcp://opcuaserver.com:4... | Browse on node 'i=85' succeeded.
20:26:39.166 | TypeCache          | opc.tcp://opcuaserver.com:4... | Reading type info of NodeId NS0|Numeric|63 succeeded
20:26:39.175 | TypeCache          | opc.tcp://opcuaserver.com:4... | Reading type info of NodeId NS0|Numeric|62 succeeded
20:26:39.181 | TypeCache          | opc.tcp://opcuaserver.com:4... | Reading type info of NodeId NS0|Numeric|2004 succeeded
20:26:39.189 | TypeCache          | opc.tcp://opcuaserver.com:4... | Reading type info of NodeId NS0|Numeric|58 succeeded
`

You may find 'used UserTokenType: Anonymous' and succeeded log result.

I used the Windows latest version of the binary.
 * https://ela.st/machinebeat-windows

Here is the DEBUG level log.

`(base) C:\Users\dwku\OneDrive\바탕 화면\machinebeat>machinebeat.exe -e -c machinebeat.yml
2020-08-11T20:39:26.690+0900    INFO    instance/beat.go:622    Home path: [C:\Users\dwku\OneDrive\바탕 화면\machinebeat] Config path: [C:\Users\dwku\OneDrive\바탕 화면\machinebeat] Data path: [C:\Users\dwku\OneDrive\바탕 화면\machinebeat\data] Logs path: [C:\Users\dwku\OneDrive\바탕  화면\machinebeat\logs]
2020-08-11T20:39:26.752+0900    DEBUG   [beat]  instance/beat.go:674    Beat metadata path: C:\Users\dwku\OneDrive\바탕 화면\machinebeat\data\meta.json
2020-08-11T20:39:26.753+0900    INFO    instance/beat.go:630    Beat ID: d51d9d17-d145-489f-b43d-f9953bc4d771
2020-08-11T20:39:26.818+0900    DEBUG   [filters]       add_cloud_metadata/providers.go:126     add_cloud_metadata: starting to fetch metadata, timeout=3s
2020-08-11T20:39:26.820+0900    DEBUG   [filters]       add_cloud_metadata/providers.go:162     add_cloud_metadata: received disposition for digitalocean after 2.0573ms. result=[provider:digitalocean, error=failed requesting digitalocean metadata: Get http://169.254.169.254/metadata/v1.json: dial tcp 169.254.169.254:80: connectex: A socket operation was attempted to an unreachable network., metadata={}]
2020-08-11T20:39:26.820+0900    DEBUG   [filters]       add_cloud_metadata/providers.go:162     add_cloud_metadata: received disposition for aws after 2.0573ms. result=[provider:aws, error=failed requesting aws metadata: Get http://169.254.169.254/2014-02-25/dynamic/instance-identity/document: dial tcp 169.254.169.254:80: connectex: A socket operation was attempted to an unreachable network., metadata={}]
2020-08-11T20:39:26.820+0900    DEBUG   [filters]       add_cloud_metadata/providers.go:162     add_cloud_metadata: received disposition for gcp after 2.0573ms. result=[provider:gcp, error=failed requesting gcp metadata: Get http://169.254.169.254/computeMetadata/v1/?recursive=true&alt=json: dial tcp 169.254.169.254:80: connectex: A socket operation was attempted to an unreachable network., metadata={}]
2020-08-11T20:39:26.821+0900    DEBUG   [filters]       add_cloud_metadata/providers.go:162     add_cloud_metadata: received disposition for openstack after 2.9711ms. result=[provider:openstack, error=failed requesting openstack metadata: Get http://169.254.169.254/2009-04-04/meta-data/instance-id: dial tcp 169.254.169.254:80: connectex: A socket operation was attempted to an unreachable network., metadata={}]
2020-08-11T20:39:26.821+0900    DEBUG   [filters]       add_cloud_metadata/providers.go:162     add_cloud_metadata: received disposition for az after 2.9711ms. result=[provider:az, error=failed requesting az metadata: Get http://169.254.169.254/metadata/instance/compute?api-version=2017-04-02: dial tcp 169.254.169.254:80: connectex: A socket operation was attempted to an unreachable network., metadata={}]
2020-08-11T20:39:26.821+0900    DEBUG   [filters]       add_cloud_metadata/providers.go:129     add_cloud_metadata: fetchMetadata ran for 2.9711ms
2020-08-11T20:39:26.821+0900    INFO    add_cloud_metadata/add_cloud_metadata.go:89     add_cloud_metadata: hosting provider type not detected.
2020-08-11T20:39:26.821+0900    DEBUG   [processors]    processors/processor.go:101     Generated new processors: add_host_metadata=[netinfo.enabled=[false], cache.ttl=[5m0s]], add_cloud_metadata=null
2020-08-11T20:39:26.822+0900    DEBUG   [seccomp]       seccomp/seccomp.go:96   Syscall filtering is only supported on Linux
2020-08-11T20:39:26.822+0900    INFO    [beat]  instance/beat.go:958    Beat info       {"system_info": {"beat": {"path": {"config": "C:\\Users\\dwku\\OneDrive\\바탕 화면\\machinebeat", "data": "C:\\Users\\dwku\\OneDrive\\바탕 화면\\machinebeat\\data", "home": "C:\\Users\\dwku\\OneDrive\\바탕 화면\\machinebeat", "logs": "C:\\Users\\dwku\\OneDrive\\바탕 화면\\machinebeat\\logs"}, "type": "machinebeat", "uuid": "d51d9d17-d145-489f-b43d-f9953bc4d771"}}}
2020-08-11T20:39:26.822+0900    INFO    [beat]  instance/beat.go:967    Build info      {"system_info": {"build": {"commit": "unknown", "libbeat": "7.6.1", "time": "1754-08-30T22:43:41.128Z", "version": "7.6.1"}}}
2020-08-11T20:39:26.823+0900    INFO    [beat]  instance/beat.go:970    Go runtime info {"system_info": {"go": {"os":"windows","arch":"amd64","max_procs":8,"version":"go1.12.4"}}}
2020-08-11T20:39:26.878+0900    INFO    [beat]  instance/beat.go:974    Host info       {"system_info": {"host": {"architecture":"x86_64","boot_time":"2020-08-11T18:46:45.8+09:00","name":"SEOUL-DKU","ip":["fe80::cc0f:37ad:34f1:3009/64","169.254.48.9/16","fe80::90e8:fdde:a2e0:8875/64","169.254.136.117/16","fe80::49f8:3301:4b72:98ad/64","192.168.0.6/24","fe80::a4fd:5499:5ae7:de0a/64","169.254.222.10/16","::1/128","127.0.0.1/8","fe80::4928:1421:1cf2:2d4b/64","172.25.129.17/28"],"kernel_version":"10.0.17134.1610 (WinBuild.160101.0800)","mac":["92:32:4b:1c:90:47","d2:32:4b:1c:90:47","90:32:4b:1c:90:47","90:32:4b:1c:90:48","72:15:91:d4:83:ca"],"os":{"family":"windows","platform":"windows","name":"Windows 10 Enterprise","version":"10.0","major":10,"minor":0,"patch":0,"build":"17134.1610"},"timezone":"KST","timezone_offset_sec":32400,"id":"e47d409c-3cca-409f-a573-66904f0b01e3"}}}
2020-08-11T20:39:26.883+0900    INFO    [beat]  instance/beat.go:1003   Process info    {"system_info": {"process": {"cwd": "C:\\Users\\dwku\\OneDrive\\바탕 화면\\machinebeat", "exe": "C:\\Users\\dwku\\OneDrive\\\ufffd\ufffd\ufffd\ufffd ȭ\ufffd\ufffd\\machinebeat\\machinebeat.exe", "name": "machinebeat.exe", "pid": 23700, "ppid": 13092, "start_time": "2020-08-11T20:39:22.592+0900"}}}
2020-08-11T20:39:26.884+0900    INFO    instance/beat.go:298    Setup Beat: machinebeat; Version: 7.6.1
2020-08-11T20:39:26.886+0900    DEBUG   [beat]  instance/beat.go:324    Initializing output plugins
2020-08-11T20:39:26.886+0900    INFO    [index-management]      idxmgmt/std.go:182      Set output.elasticsearch.index to 'machinebeat-7.6.1' as ILM is enabled.
2020-08-11T20:39:26.887+0900    INFO    elasticsearch/client.go:174     Elasticsearch url: http://localhost:9200
2020-08-11T20:39:26.888+0900    DEBUG   [publisher]     pipeline/consumer.go:137        start pipeline event consumer
2020-08-11T20:39:26.889+0900    INFO    [publisher]     pipeline/module.go:110  Beat name: SEOUL-DKU
2020-08-11T20:39:26.892+0900    DEBUG   [modules]       beater/metricbeat.go:155        Available modules and metricsets: Register [ModuleFactory:[beat, docker, elasticsearch, kibana, logstash, mongodb, mysql, postgresql, system, uwsgi, windows], MetricSetFactory:[aerospike/namespace, apache/status, beat/state, beat/stats, ceph/cluster_disk, ceph/cluster_health, ceph/cluster_status, ceph/monitor_health, ceph/osd_df, ceph/osd_tree, ceph/pool_disk, consul/agent, couchbase/bucket, couchbase/cluster, couchbase/node, couchdb/server, docker/container, docker/cpu, docker/diskio, docker/event, docker/healthcheck, docker/image, docker/info, docker/memory, docker/network, dropwizard/collector, elasticsearch/ccr, elasticsearch/cluster_stats, elasticsearch/enrich, elasticsearch/index, elasticsearch/index_recovery, elasticsearch/index_summary, elasticsearch/ml_job, elasticsearch/node, elasticsearch/node_stats, elasticsearch/pending_tasks, elasticsearch/shard, envoyproxy/server, etcd/leader, etcd/metrics, etcd/self, etcd/store, golang/expvar, golang/heap, graphite/server, haproxy/info, haproxy/stat, http/json, http/server, jolokia/jmx, kafka/consumergroup, kafka/partition, kibana/stats, kibana/status, kubernetes/apiserver, kubernetes/container, kubernetes/controllermanager, kubernetes/event, kubernetes/node, kubernetes/pod, kubernetes/proxy, kubernetes/scheduler, kubernetes/state_container, kubernetes/state_cronjob, kubernetes/state_deployment, kubernetes/state_node, kubernetes/state_persistentvolume, kubernetes/state_persistentvolumeclaim, kubernetes/state_pod, kubernetes/state_replicaset, kubernetes/state_resourcequota, kubernetes/state_service, kubernetes/state_statefulset, kubernetes/system, kubernetes/volume, kvm/dommemstat, logstash/node, logstash/node_stats, memcached/stats, mongodb/collstats, mongodb/dbstats, mongodb/metrics, mongodb/replstatus, mongodb/status, mqtt/topic, munin/node, mysql/galera_status, mysql/status, nats/connections, nats/routes, nats/stats, nats/subscriptions, nginx/stubstatus, opcua/nodevalue, php_fpm/pool, php_fpm/process, postgresql/activity, postgresql/bgwriter, postgresql/database, postgresql/statement, prometheus/collector, rabbitmq/connection, rabbitmq/exchange, rabbitmq/node, rabbitmq/queue, redis/info, redis/key, redis/keyspace, system/core, system/cpu, system/diskio, system/filesystem, system/fsstat, system/memory, system/network, system/network_summary, system/process, system/process_summary, system/raid, system/service, system/socket_summary, system/uptime, traefik/health, uwsgi/status, vsphere/datastore, vsphere/host, vsphere/virtualmachine, windows/perfmon, windows/service, zookeeper/connection, zookeeper/mntr, zookeeper/server], LightModules:[LightModules:[]]]
2020-08-11T20:39:26.895+0900    INFO    instance/beat.go:439    machinebeat start running.
2020-08-11T20:39:26.895+0900    INFO    [monitoring]    log/log.go:118  Starting metrics logging every 30s
2020-08-11T20:39:26.895+0900    DEBUG   [service]       service/service_windows.go:72   Windows is interactive: true
2020-08-11T20:39:26.896+0900    DEBUG   [cfgfile]       cfgfile/reload.go:133   Checking module configs from: C:\Users\dwku\OneDrive\바탕 화면\machinebeat/modules.d/*.yml
2020-08-11T20:39:26.899+0900    DEBUG   [cfgfile]       cfgfile/cfgfile.go:193  Load config from file: C:\Users\dwku\OneDrive\바탕 화면\machinebeat\modules.d\opcua.yml
2020-08-11T20:39:26.907+0900    DEBUG   [cfgfile]       cfgfile/reload.go:147   Number of module configs found: 1
2020-08-11T20:39:26.907+0900    WARN    [cfgwarn]       nodevalue/nodevalue.go:92       BETA: The OPCUA metricset is beta.
2020-08-11T20:39:26.908+0900    INFO    nodevalue/client.go:54  [OPCUA] Get all endpoints from opc.tcp://host.docker.internal:53530/OPCUA/SimulationServer
2020-08-11T20:39:26.929+0900    INFO    nodevalue/client.go:68  [OPCUA] Policy URI: http://opcfoundation.org/UA/SecurityPolicy#None with security mode MessageSecurityModeNone
2020-08-11T20:39:26.929+0900    INFO    nodevalue/client.go:88  [OPCUA] Set to anonymous login
2020-08-11T20:39:27.941+0900    ERROR   nodevalue/nodevalue.go:165      [OPCUA] Tried to connect to OPC UA server 1 time(s). Without success.
2020-08-11T20:39:27.943+0900    INFO    [monitoring]    log/log.go:153  Total non-zero metrics  {"monitoring": {"metrics": {"beat":{"cpu":{"system":{"ticks":140,"time":{"ms":140}},"total":{"ticks":249,"time":{"ms":249},"value":249},"user":{"ticks":109,"time":{"ms":109}}},"handles":{"open":233},"info":{"ephemeral_id":"80fff009-f2f1-430f-9287-47c381f328d5","uptime":{"ms":5324}},"memstats":{"gc_next":5858016,"memory_alloc":3367376,"memory_total":6717624,"rss":34832384},"runtime":{"goroutines":17}},"libbeat":{"config":{"module":{"running":0}},"output":{"type":"elasticsearch"},"pipeline":{"clients":0,"events":{"active":0}}},"system":{"cpu":{"cores":8}}}}}
2020-08-11T20:39:27.944+0900    INFO    [monitoring]    log/log.go:154  Uptime: 5.3277362s
2020-08-11T20:39:27.945+0900    INFO    [monitoring]    log/log.go:131  Stopping metrics logging.
2020-08-11T20:39:27.947+0900    INFO    instance/beat.go:445    machinebeat stopped.
2020-08-11T20:39:27.947+0900    ERROR   instance/beat.go:933    Exiting: 1 error: Connection was not possible
Exiting: 1 error: Connection was not possible`


Thank you!

**Comment by felix-lessoer** [comment]
I was investigating into this issue today.
The problem is, that the prosys server is failing when ApplicationName is not set during the session creation. 
This is the first server that acts like this. 

Unfortunatly the underlying OPC UA library for that beat is also failing when I try to set the ApplicationName. So I need to fix this first to work around the issue in the prosys server.

**Comment by dongwoo-john-ku** [comment]
Hello @felix-lessoer,

Thank you for your kind answer.
Please let me know when you release the updated version.

**Comment by felix-lessoer** [comment]
@dongwoo-john-ku Prosys released a new version of their server yesterday. The issue that prevents connecting has been fixed now. So connection can be established with that version.

Unfortunatly it looks like this server implementation has more errors. I got invalid data when browsing. I didn't test to directly read the value from specific nodes. That may works.

But in general I would recommend using a better implementation of an OPC UA server.
